bemani/backend/gf/gfv8.py

```diff
@@ -0,0 +1,2657 @@
+
+    def db_to_game_chart(self, db_chart: int) -> int:
+        return {
+            self.CHART_TYPE_GUITAR_BEGINNER: self.GAME_CHART_TYPE_GUITAR_BEGINNER,
+            self.CHART_TYPE_GUITAR_BASIC: self.GAME_CHART_TYPE_GUITAR_BASIC,
+            self.CHART_TYPE_GUITAR_ADVANCED: self.GAME_CHART_TYPE_GUITAR_ADVANCED,
+            self.CHART_TYPE_GUITAR_EXTREME: self.GAME_CHART_TYPE_GUITAR_EXTREME,
+            self.CHART_TYPE_BASS_BEGINNER: self.GAME_CHART_TYPE_BASS_BEGINNER,
+            self.CHART_TYPE_BASS_BASIC: self.GAME_CHART_TYPE_BASS_BASIC,
+            self.CHART_TYPE_BASS_ADVANCED: self.GAME_CHART_TYPE_BASS_ADVANCED,
+            self.CHART_TYPE_BASS_EXTREME: self.GAME_CHART_TYPE_BASS_EXTREME,
+            self.CHART_TYPE_OPEN_BEGINNER: self.GAME_CHART_TYPE_OPEN_BEGINNER,
+            self.CHART_TYPE_OPEN_BASIC: self.GAME_CHART_TYPE_OPEN_BASIC,
+            self.CHART_TYPE_OPEN_ADVANCED: self.GAME_CHART_TYPE_OPEN_ADVANCED,
+            self.CHART_TYPE_OPEN_EXTREME: self.GAME_CHART_TYPE_OPEN_EXTREME,
+        }[db_chart]
+
+    def handle_shopinfo_regist_request(self, request: Node) -> Node:
+        # fuck
+        self.update_machine_name(request.child_value('shop/name'))
+
+        shopinfo = Node.void('shopinfo')
+
+        data = Node.void('data')
+        shopinfo.add_child(data)
+        data.add_child(Node.u32('cabid', 1))
+        data.add_child(Node.string('locationid', 'LOCATION'))
+        data.add_child(Node.u8('is_send', 1))
+
+        return shopinfo
+
+    def handle_demodata_get_request(self, request: Node) -> Node:
+        root = Node.void('demodata')
+        mode = Node.u8('mode', 0)
+        root.add_child(mode)
+
+        hitchart = Node.void('hitchart')
+        root.add_child(hitchart)
+        hitchart.set_attribute('nr', '0')
+
+        hitchart.add_child(Node.string('start', "0"))
+        hitchart.add_child(Node.string('end', "0"))
+
+
+        bossdata = Node.void('bossdata')
+        root.add_child(bossdata)
+
+        bossdata.add_child(Node.u8('division', 15))
+        border = Node.u8_array(
+            'border',
+            [
+                0,
+                0,
+                0,
+                0,
+                0,
+                0,
+                0,
+                0,
+                0,
+            ]
+
+        )
+
+        bossdata.add_child(border)
+        bossdata.add_child(Node.u8('extra_border', 90))
+        bossdata.add_child(Node.u8('bsc_encore_border', 90))
+        bossdata.add_child(Node.u8('adv_encore_border', 91))
+        bossdata.add_child(Node.u8('ext_encore_border', 92))
+        bossdata.add_child(Node.u8('bsc_premium_border', 93))
+        bossdata.add_child(Node.u8('adv_premium_border', 93))
+        bossdata.add_child(Node.u8('ext_premium_border', 93))
+
+        info = Node.void('info')
+        root.add_child(info)
+
+        info.add_child(Node.string('message', "Brought to you by PhaseII! Hosted on phaseii.iidxfan.xyz. Login today!"))
```

Comment:
Dude really? This is about as bad as bundle sites advertising in the zip file where you downloaded somebody else's software.